test/http-test.js

```diff
@@ -338,6 +341,49 @@ describe('HTTP', function() {
     }
   });
 
+  it('should generate 10 blocks from RPC call', async () => {
+    blocks = await nclient.execute(
+      'generatetoaddress',
+      [10, addr.toString('regtest')]
+    );
+    assert.strictEqual(blocks.length, 10);
+  });
+
+  it('should initiate rescan from socket without a bloom filter', async () => {
+    // Rescan from height 5. Without a filter loaded = no response, but no error
+    const response = await nclient.call('rescan', 5);
+    assert.strictEqual(null, response);
+  });
+
+  it('should initiate rescan from socket WITH a bloom filter', async () => {
+    // Create an SPV-standard Bloom filter and add one of our wallet addresses
+    const filter = BloomFilter.fromRate(20000, 0.001, BloomFilter.flags.ALL);
+    filter.add(addr.toString('regtest'), 'ascii');
+
+    // Send Bloom filter to server
+    await nclient.call('set filter', filter.filter);
+
+    // Rescan is a weird call, the server sends a socket event BACK to the
+    // client after every block with a matching transaction. Without a hook in
+    // place in the CLIENT, the rescan call will result in the error:
+    // "Call not found: block rescan."
+    // This design allows for the client to handle the wallet. Rescan is a NODE
+    // call and the server is not running the wallet.
+    const matchingBlocks = [];
+    nclient.hook('block rescan', (entry, txs) => {
+      // Coinbase transactions were mined to our watch address, matches filter.
+      assert.strictEqual(txs.length, 1);
+      matchingBlocks.push(
+        // Blocks are returned as raw ChainEntry
+        ChainEntry.fromRaw(entry).rhash().toString('hex')
+      );
+    });
+
+    // Rescan from height 5 -- should return blocks 5 through 10, inclusive.
+    await nclient.call('rescan', 5);
+    assert.deepStrictEqual(matchingBlocks, blocks.slice(4));
```

Comment:
Could match the txs for a stricter test case.

Reply:
Added assertion to test the transaction output address is correct.